File: desktop/src/types/mention.ts
import type { User } from "./user";
import type { Message } from "./message";

export interface UserGroup {
  id: string;
  name: string;
  description: string;
  creator_id: string;
  created_at: string;
  updated_at: string;
  members?: User[];
}

export interface Mention {
  id: string;
  message_id: string;
  channel_id: string;
  mentioned_user_id?: string;
  mentioned_group_id?: string;
  mention_type: "user" | "group" | "channel" | "here";
  is_read: boolean;
  created_at: string;
  message?: Message;
}

File: server/internal/model/mention.go
package model

import (
	"time"

	"github.com/google/uuid"
)

type UserGroup struct {
	ID          uuid.UUID `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	CreatorID   uuid.UUID `json:"creator_id"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
	Members     []User    `json:"members,omitempty"`
}

type CreateUserGroupRequest struct {
	Name        string `json:"name" validate:"required,min=2,max=100"`
	Description string `json:"description" validate:"max=500"`
}

type UpdateUserGroupRequest struct {
	Name        *string `json:"name" validate:"omitempty,min=2,max=100"`
	Description *string `json:"description" validate:"omitempty,max=500"`
}

type GroupMemberRequest struct {
	UserID uuid.UUID `json:"user_id" validate:"required"`
}

type Mention struct {
	ID               uuid.UUID  `json:"id"`
	MessageID        uuid.UUID  `json:"message_id"`
	ChannelID        uuid.UUID  `json:"channel_id"`
	MentionedUserID  *uuid.UUID `json:"mentioned_user_id,omitempty"`
	MentionedGroupID *uuid.UUID `json:"mentioned_group_id,omitempty"`
	MentionType      string     `json:"mention_type"`
	IsRead           bool       `json:"is_read"`
	CreatedAt        time.Time  `json:"created_at"`
	Message          *Message   `json:"message,omitempty"`
}

type MarkMentionsReadRequest struct {
	ChannelID uuid.UUID `json:"channel_id" validate:"required"`
}
